sys/kern/mmap.c

```diff
@@ -60,18 +60,29 @@ int do_munmap(vaddr_t addr, size_t length) {
 
   vm_map_t *uspace = proc_self()->p_uspace;
 
+  if (length == 0)
+    return EINVAL;
+
+  /* Round boundaries to contain entire pages from range [addr, addr + length)
+   */
+  vaddr_t right_boundary = roundup(addr + length, PAGESIZE);
+  addr = rounddown(addr, PAGESIZE);
```

Comment:
The address should be already page aligned, check the manual.

Reply:
We should be more strict here and return `EINVAL` in case of wrong alignment.

Reply:
Hmm the POSIX says:

```
The munmap() function may fail if:

[EINVAL]
The addr argument is not a multiple of the page size as returned by sysconf().
```

"may", not "have to", so I don't really know, what is the best solution.

Reply:
So could you give **any** example of **real** program on **real** OS that doesn't fail on that condition?
We could implement POSIX in every detail but I don't see any reason for that. This will be a **dead** feature. We don't implement programs in user space - we only care about kernel so nobody will use that combination of parameters except our own tests. Do we have tests for that?

Reply:
Changed.